### ISSUE TITLE
provide an advanced DB interface for internals

### ISSUE THREAD

**Issue by jchappelow** [kwilteam/kwil-db#800]
The current DB methods scan in all rows at once and into `[][]any` since when dealing with user-authored queries we have no idea what concrete Go type to *scan* into.

However, there are many cases in internal code, including bookkeeping business of the engine, where it would be more straightforward and less error prone (we pick the type) if there were methods for this purpose.

This could be accomplished with a new Execish method that takes scan targets, or it could be done the traditional way with a row iterator (next, scan, err, etc).

**Comment by jchappelow** [comment]
A specific case where this is problematic is with the `uint256` type which is domain and posgresql claims the OID of the underlying type when scanning it out.

**Comment by jchappelow** [comment]
Related note, I've been making functions and methods to do said instantiation and scanning to support statistics collections.  We could expose some of that or even have the engine use that approach rather than relying so heavily on the pgx type map and then converting the `[]any` from `row.Values()`.

Related comment on the issue of efficiency: https://github.com/kwilteam/kwil-db/pull/837#discussion_r1645167702

In short, I indented these "advanced" features with caller instantiated scan variables to support internals like our stores, but user datasets could go this route if the engine instantiates the scan variables, typed based on the schema or just the column info from `information_schema.columns` when the select/projection is a simple column reference (otherwise an `any`).
